### DISCUSSION TITLE
How do use the optimized to predict the test_data?

### DISCUSSION THREAD

**Discussion by LittleAnt69** [ourownstory/neural_prophet#1491] [Feature Requests & Feedback]
When I train the model, assuming epoch=100, when epoch=50, the mse is the smallest. So how do I use the optimized model to predict the test_data? 
In addition, the parameter checkpointing=True can be set in the fit() function to save the optimized  model parameters, but I have no way to load this model parameter. Is there any way to load the saved parameters? 
The parameter form is as follows：epoch=3-step=92.ckpt.

**Comment by SimonWittner** [comment]
Hey, 
I am unsure if I fully understand your issue. To use an optimized model with 50 epochs, you can define your model with new hyperparameters with: `m = NeuralProphet(epochs=50)`. 
To my knowledge checkpointing is not fully implemented and will be a feature for later versions.

**Reply by LittleAnt69** [reply]
When I train the model, I don't know what epoch equals when the loss is the smallest. So I set epoch=100. After training, I found that when epoch=50, the loss was smallest. So how do I use epoch=50 to test new data? What I mean is how to save and load the intermediate optimal model? Do I define and train a new model again：m = NeuralProphet(epochs=50)?
